app/drivers/distributed_manager.go

```diff
@@ -33,6 +33,42 @@ func NewDistributedManager(manager *Manager, outboxStore store.OutboxStore) *Dis
 	}
 }
 
+// Provision returns an instance for a job execution and tags it as in use.
+// This method and BuildPool method contain logic for maintaining pool size.
+func (d *DistributedManager) ReserveCapacity(
+	ctx context.Context,
+	poolName,
+	serverName,
+	ownerID,
+	resourceClass string,
+	vmImageConfig *spec.VMImageConfig,
+	query *types.QueryParams,
+	storageConfig *types.StorageConfig,
+	zone string,
+	machineType string,
+	shouldUseGoogleDNS bool,
+	timeout int64,
+) (*types.CapacityReservation, bool, error) {
+	pool, err := d.validatePool(poolName)
+	if err != nil {
+		return nil, false, err
+	}
+	return d.reserveCapacityFromPool(
+		ctx,
+		pool,
+		serverName,
+		ownerID,
+		resourceClass,
+		vmImageConfig,
+		storageConfig,
+		zone,
+		machineType,
+		shouldUseGoogleDNS,
+		timeout,
+		poolName,
+	)
```

Comment:
I think this should in turn call Provision method only since the logic is mostly same.